internal/clusterconfig/clusterconfig.go

```diff
@@ -295,3 +307,64 @@ func (r *UpgradeClusterConfigGather) getIDMSs(ctx context.Context) (v1.ImageDige
 
 	return idmsList, nil
 }
+
+func (r *UpgradeClusterConfigGather) fetchICSPs(ctx context.Context, manifestsDir string) error {
+	iscpsList := &v1.ImageContentPolicyList{}
+	currentIcps := &v1.ImageContentPolicyList{}
+	if err := r.Client.List(ctx, currentIcps); err != nil {
+		return err
+	}
+
+	for _, icp := range currentIcps.Items {
+		obj := v1.ImageContentPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: icp.ObjectMeta.Name,
+			},
+			Spec: icp.Spec,
+		}
+		typeMeta, err := r.typeMetaForObject(&icp)
+		if err != nil {
+			return err
+		}
+		icp.TypeMeta = *typeMeta
+		iscpsList.Items = append(iscpsList.Items, obj)
+	}
+	typeMeta, err := r.typeMetaForObject(iscpsList)
+	if err != nil {
+		return err
+	}
+	iscpsList.TypeMeta = *typeMeta
+
+	if err := utils.WriteToFile(iscpsList, filepath.Join(manifestsDir, icspsFileName)); err != nil {
+		return fmt.Errorf("failed to write icsps to %s, err: %w",
+			filepath.Join(manifestsDir, icspsFileName), err)
+	}
+
+	return nil
+}
+
+func (r *UpgradeClusterConfigGather) fetchCABundle(ctx context.Context, manifestsDir string) error {
+	r.Log.Info("Fetching user ca bundle")
+	caBundle := &corev1.ConfigMap{}
+	err := r.Client.Get(ctx, types.NamespacedName{Name: caBundleCMName,
+		Namespace: configNamespace}, caBundle)
+	if err != nil && errors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("failed to get ca bundle cm, err: %w", err)
+	}
```

Comment:
Perhaps we can replace these lines with the function `GetConfigMap` in [func GetConfigMap(ctx context.Context, c client.Client, configMap v1alpha1.ConfigMapRef) (*corev1.ConfigMap, error) {](https://github.com/openshift-kni/lifecycle-agent/blob/main/internal/common/utils.go#L32-L43).

Reply:
I can, just not fully understand why we actually have this util function and what it gives us. Will be glad to understand.

Reply:
Sure! This function, was defined and, is used by several plugins. The intent is to define commonly used functions in a unified location that can be easily referenced. In your case, you only need to fetch a ConfigMap once, however, there are cases where ConfigMaps need to be retrieved in several places - thus the reasoning behind its existence (avoid code duplication).